### PR TITLE
[UI v2] feat: Adding generic empty-state component, and replaced variables empty-state for it

### PR DESCRIPTION
**Pull request by devinvillarosa**
Looking at the upcoming concurrency page, there can be some generic components that can be re-used. One of them being `EmptyState`. This PR:
1. Adds a generic `EmptyState` composite component. This components manages the styling and pieces needed to create an Empty State
2. Replaces variables EmptyState to use this generic `EmptyState`

Screenshot 2024-12-03 at 10 29 25 AM
Screenshot 2024-12-03 at 11 47 27 AM

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [x] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [x] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: ui-v2/.gitignore [hunk 1]
```diff
@@ -28,3 +28,4 @@ dist-ssr
 oss_schema.json
 *.tsbuildinfo
 *storybook.log
+*.orig
```
Comment (devinvillarosa) [inline ui-v2/.gitignore:31]: this is generated when running `git mergetool` (with vscode)

File: ui-v2/src/components/ui/empty-state.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		<CardContent className="flex flex-col gap-2 items-center justify-center py-16">
+			{children}
+		</CardContent>
+	</Card>
+);
+
+EmptyState.Icon = EmptyStateIcon;
+EmptyState.Title = EmptyStateTitle;
+EmptyState.Description = EmptyStateDescription;
+EmptyState.Actions = EmptyStateActions;
```
Comment (desertaxle) [inline ui-v2/src/components/ui/empty-state.tsx:40]: This is similar to the pattern the `shadcn/ui` components we're using, but those modules will export each component individually (like in `ui-v2/src/components/ui/card.tsx`). I think we should be consistent with how we expose sets of collaborative components. We could update our other components to follow this pattern, but I'd prefer to follow the `shadcn/ui` style, so we don't diverge too much from the `shadcn/ui` source. Let me know if you feel strongly the other way and we can discuss!

**Review by desertaxle — COMMENTED** [review]
Nice! I have one comment on the export method for these new components, but this makes a lot of sense as a way to organize the layout.

I'm going back and forth on if it's worth it to write tests for these new components. What do you think?

File: ui-v2/src/components/ui/empty-state.tsx (hunk 2 above)
Comment (devinvillarosa) [reply ui-v2/src/components/ui/empty-state.tsx:40]: sounds good. Let me make an update

Comment (devinvillarosa) [reply ui-v2/src/components/ui/empty-state.tsx]: Updated in latest push

**Comment by devinvillarosa** [comment]
> Nice! I have one comment on the export method for these new components, but this makes a lot of sense as a way to organize the layout.
> 
> I'm going back and forth on if it's worth it to write tests for these new components. What do you think?

Gut feeling, it seems like this is more of a UI only type of component.
And we can test its functionality coverage when we write something close to `variables.test.tsx`, or even a Playwright test.

For UI testing, maybe we can introduce browser screenshot testing in the future like [Happo](https://happo.io/).

**Review by desertaxle — APPROVED** [review]

**Pull request merged**